Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        # complications arise from having a completely different signature from pandas.
+
+        # is pandas is calling this as part of its transformations, pass it through
+        all_kwargs_are_pandas = len(set(kwargs.keys()).difference(cls._KNOWN_PANDAS_KWARGS)) == 0
+
+        ##### START PANDAS SUBCLASS COMPATIBILITY SECTION #####
+        # this compatibility hack must exist because we override __new__ in a way incompatible
+        # with pandas' subclassing scheme---hacks are needed for compatibility, read on
+
+        # if we called the constructor without arguments, create empty
+        # all possible pandas kwargs will be passed to __init__, where
+        # pandas will be able to process them
+        if not args and (not kwargs or all_kwargs_are_pandas):
+            return super().__new__(cls)
+
+        # pandas can call this constructor (through Table._constructor) internally
+        # when doing transformations: its signatures are either a BlockManager or dict as the sole arg,
+        # or a ndarray as the sole arg, accompanied by at least one kwarg
+        # this serves the purpose of transforming the first arg into its matching kwarg,
+        # which will then be processed by the if clause a couple of lines above this
+        if len(args) == 1 and (isinstance(args[0], (pd.core.internals.BlockManager, dict))
+                               or (isinstance(args[0], np.ndarray) and len(kwargs) != 0 and all_kwargs_are_pandas)):
+            return cls(data=args[0], **kwargs)
+        ##### END PANDAS SUBCLASS COMPATIBILITY SECTION #####
+
+        if 'filename' in kwargs:
+            args = [kwargs.pop('filename')]
+
+        if isinstance(args[0], str):
+            if args[0].startswith('https://') or args[0].startswith('http://'):
+                return cls.from_url(args[0], **kwargs)
+            else:
+                return cls.from_file(args[0], **kwargs)
+        elif isinstance(args[0], TableBase):
+            return cls.from_table(args[0].domain, args[0])
+        elif isinstance(args[0], pd.DataFrame):
+            return cls.from_dataframe(None, args[0], **kwargs)
+        elif isinstance(args[0], Domain):
+            domain, args = args[0], args[1:]
+            if not args:
+                return cls.from_domain(domain, **kwargs)
+            if isinstance(args[0], TableBase):
+                return cls.from_table(domain, *args)
+            elif isinstance(args[0], list):
+                return cls.from_list(domain, *args)
+            elif isinstance(args[0], pd.DataFrame):
+                return cls.from_dataframe(domain, args[0], **kwargs)
+        else:
+            domain = None
+        return cls.from_numpy(domain, *args, **kwargs)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the object.
+
+        Sets the name and other attributes and calls pandas' initialization.
+        Filters the domain: retains only the columns actually in the table.
```

Comment:
That the domain of iris[["sepal length"]] should only be [sepal length]. It actually doesn't work currently, thanks for pointing this out. I think I have a better solution than putting this in the constructor. Better to discuss it in person.